swagger.yaml

```diff
@@ -3152,83 +3152,3050 @@ paths:
+        "500":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "503":
+          description: Unavailable.
+          schema:
+            $ref: '#/definitions/error'
+
+  /v2/infra-envs/{infra_env_id}:
+    get:
+      tags:
+        - installer
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - agentAuth: []
+      description: Retrieves the details of the InfraEnv.
+      operationId: GetInfraEnv
+      parameters:
+        - in: path
+          name: infra_env_id
+          description: The InfraEnv to be retrieved.
+          type: string
+          format: uuid
+          required: true
+      responses:
+        "200":
+          description: Success.
+          schema:
+            $ref: '#/definitions/infra_env'
+        "401":
+          description: Unauthorized.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "403":
+          description: Forbidden.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "404":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "405":
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
+        "500":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "503":
+          description: Unavailable.
+          schema:
+            $ref: '#/definitions/error'
+
+    patch:
+      tags:
+        - installer
+      description: Updates an InfraEnv.
+      operationId: UpdateInfraEnv
+      parameters:
+        - in: path
+          name: infra_env_id
+          description: The InfraEnv to be updated.
+          type: string
+          format: uuid
+          required: true
+        - in: body
+          name: infra-env-create-params
+          description: The properties to update.
+          required: true
+          schema:
+            $ref: '#/definitions/infraenv-create-params'
+      responses:
+        "201":
+          description: Success.
+          schema:
+            $ref: '#/definitions/infra_env'
+        "400":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "401":
+          description: Unauthorized.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "403":
+          description: Forbidden.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "404":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "405":
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
+        "409":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "500":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+
+    delete:
```

Comment:
what happen to hosts related to deleted infra env? (probably a question related to late binding and not the API but asking anyway)

Reply:
I assume that we want to delete the agents resource as well since it's subresource to infraenv, on the host itself I don't think that something will change until someone will boot it with a new iso

Reply:
Good question.  Maybe we want to block deletion as long as Agents exist?

Reply:
And if they are attached to a cluster?

Reply:
can you delete today's hosts from installed/not-installed cluster?
I guess it should be the same.